### PR TITLE
added class_weights for cross entropy loss to segmentation.py

### PR DESCRIPTION
**Pull request by nsutezo**
Adding ability to pass in class weights into cross entropy loss.

### CODE REVIEW

**Review by calebrob6 — APPROVED** [review]

File: torchgeo/trainers/segmentation.py [hunk 1]
```diff
@@ -84,6 +91,8 @@ def __init__(self, **kwargs: Any) -> None:
                 the backbone
             in_channels: Number of channels in input image
             num_classes: Number of semantic classes to predict
+            class_weights: Optional rescaling weight given to each
+                class and used with 'ce' loss
             loss: Name of the loss function, currently supports
                 'ce', 'jaccard' or 'focal' loss
             ignore_index: Optional integer class index to ignore in the loss and metrics
```
Comment (adamjstewart) [inline torchgeo/trainers/segmentation.py:98]: Can you add:
```rst
.. versionadded: 0.5
   The *class_weights* parameter.
```
to the end of the docstring?

**Review by adamjstewart — COMMENTED** [review]
Should we also add this to the classification and detection trainers as well?

File: torchgeo/trainers/segmentation.py [hunk 2]
```diff
@@ -100,6 +109,8 @@ def __init__(self, **kwargs: Any) -> None:
            The *segmentation_model* parameter was renamed to *model*,
            *encoder_name* renamed to *backbone*, and
            *encoder_weights* to *weights*.
```
Comment (adamjstewart) [inline torchgeo/trainers/segmentation.py:111]: Prob need a newline between sections

File: torchgeo/trainers/segmentation.py [hunk 3]
```diff
@@ -115,6 +126,11 @@ def __init__(self, **kwargs: Any) -> None:
                 UserWarning,
             )
         self.ignore_index = kwargs["ignore_index"]
+
+        self.class_weights = kwargs.get("class_weights", None)
+        if not isinstance(self.class_weights, (list, type(None))):
+            raise ValueError("class_weights must be a List or None")
```
Comment (adamjstewart) [inline torchgeo/trainers/segmentation.py:132]: I prob wouldn't bother with manual type checking, that's what mypy is for

Comment (nsutezo) [reply torchgeo/trainers/segmentation.py:132]: Is it okay if I take out lines 131 & 132?

Comment (adamjstewart) [reply torchgeo/trainers/segmentation.py:132]: Yep

**Review by adamjstewart — APPROVED** [review]
LGTM. Only remaining question is if we want to do this for classification/detection trainers too or save that for a different PR?

**Comment by calebrob6** [comment]
Different PR

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Thanks for the PR!

**Pull request merged**